src/poetry/utils/env.py

```diff
@@ -1541,7 +1541,7 @@ def script_dirs(self) -> list[Path]:
                 self._script_dirs.append(self.userbase / self._script_dirs[0].name)
         return self._script_dirs
 
-    def _bin(self, bin: str) -> Path:
+    def _bin(self, bin: str) -> str:
```

Comment:
Maybe, we should mention why we can't return a `Path` in the docstring.

Reply:
suggestions welcome.  I find this such a confused function that I don't know how to write a sensible sentence describing it!

Reply:
```
Return path to the given executable as required by subprocess.POpen().

Caution: We can't return Path objects because we need to preserve
the leading "./" in relative paths (e.g. if "bin" is a shell script).
```

Reply:
seems to me that belongs in `get_command_from_bin()` if anywhere; and that the testcase more than has it covered.  

Eg an alternative fix that I considered was for `_bin()` to continue to return a `Path` but actually look for the executable with `shutil.which()` - though then you have to account for the case where it doesn't exist earlier than it is currently done.

Reply:
Yeah, probably the test case is enough. I'm just wondering if the test case needs some explanation? What if someone thinks `assert command == ["foo.py"]` is just as good as `assert command == ["./foo.py"]`? Both paths are not absolute paths...

Reply:
Of course testcases are always vulnerable to someone deciding they were wrong all along, but this one is called `test_command_from_bin_preserves_relative_path()`.  

If that's not enough to give pause for thought when removing the './' - I would expect at that point to `git blame` and find this discussion - then I doubt that putting a comment in the code under test is going to stop them.

I'll add an link from the testcase to the original issue, maybe that'll save someone a `git blame` one day.

Reply:
Probably, I'm overthinking this. We'll see. Thank you.